Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp

```diff
@@ -159,22 +175,33 @@ void CredentialRequestCoordinator::presentPicker(CredentialPromise&& promise, Di
             if (!weakThis)
                 return;
             LOG(DigitalCredentials, "Credential picker was aborted by AbortSignal");
-            weakThis->abortPicker(reason);
+            weakThis->abortPicker(WTFMove(reason));
         });
     }
 
     setState(PickerState::Presenting);
     setCurrentPromise(WTFMove(promise));
+    observeContext(document.scriptExecutionContext());
 
     auto weakThis = WeakPtr { *this };
     m_client->showDigitalCredentialsPicker(
+        WTFMove(unvalidatedRequests),
         requestData,
         [weakThis = WeakPtr { *this }, signal](Expected<DigitalCredentialsResponseData, ExceptionData>&& responseOrException) {
             if (RefPtr protectedThis = weakThis.get())
                 protectedThis->handleDigitalCredentialsPickerResult(WTFMove(responseOrException), signal);
         });
 }
 
+ExceptionOr<DigitalCredentialsRequestData> CredentialRequestCoordinator::prepareDigitalCredentialRequestData(const Document& document, Vector<ValidatedDigitalCredentialRequest>&& requests)
+{
+    DigitalCredentialsRequestData requestData;
+    requestData.topOrigin = document.protectedTopOrigin()->data().isolatedCopy();
```

Comment:
Why the isolatedCopy()?

Reply:
copy/pasta... I'll change it.

Reply:
Actually, I used that because it said:
```
    // Create a deep copy of this SecurityOrigin. This method is useful
    // when marshalling a SecurityOrigin to another thread.
    WEBCORE_EXPORT Ref<SecurityOrigin> isolatedCopy() const;
```

Is `crossThreadCopy(document.protectedTopOrigin()->data())` the right thing to do here?

Reply:
It is not clear to me how multi-threading is involved here. Normally we call `crossThreadCopy()` / `isolatedCopy()` when dispatching to another thread. Calling them inside a function like this that is not named isolatedCopy() and doesn't dispatch to another thread is unexpected, thus my comment.

Reply:
Right, I'd misunderstood the intent of `.isolatedCopy()`. Hopefully all good now.